core/embed/trezorhal/unix/button.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include TREZOR_BOARD
+
+#ifdef USE_BUTTON
```

Comment:
could we remove this ifdef in favor of only compiling this file for 'board' that support buttons? (likewise for USE_TOUCH)

Reply:
I've just removed these ifdefs, see 31b0bf98500a63a40288b8e7a441be82edd27394. Any changes in the build script are  not necessary.

Reply:
Not necessary for sure, but should lower the compilation time both locally and also in CI, where we build a lot (though it remains very questionable whether it has some real effect), and its more consistent with how it is done in HW builds

Reply:
I agree with this, but I believe such a change requires some preparation. Currently, `USE_TOUCH` and `USE_BUTTON` are defined in header files, so we cannot use them in build scripts. This situation suggests we should open a discussion about the consistent use of feature flags across C, Rust, and build scripts.

Reply:
The build scripts are actually quite ready for this, see https://github.com/trezor/trezor-firmware/commit/7a8e2535c0ab930a8ea7bfcd7ff8744392f0fd5f - feel free to cherrypick. 

Yes we could also define `USE_TOUCH` and `USE_BUTTON` there or discuss some other solutions, but its not needed for this particular change.

Reply:
Nice, I didn't know about this. Cherry picked: 50f67d53db1934e2df49ddcc923e87ccbd96419e